test/games/strategy/engine/framework/map/download/DownloadFileParserTest.java

```diff
@@ -22,53 +19,53 @@
   public void testParseMap() {
     final ByteArrayInputStream inputStream = new ByteArrayInputStream(buildTestXml());
     List<DownloadFileDescription> games = DownloadFileParser.parse(inputStream);
-    assertThat(games.size(), is(4));
+    assertEquals(4, games.size());
     DownloadFileDescription desc = games.get(0);
-    assertThat(desc.getUrl(), is("http://example.com/games/game.zip"));
-    assertThat(desc.getDescription(), Matchers.containsString("Some notes"));
-    assertThat(desc.getMapName(), is(GAME_NAME));
+    assertEquals(desc.getUrl(), "http://example.com/games/game.zip");
```

Comment:
Notice that the parameters are now flipped.

Check the assertEquals API, it expects the "expected" parameter first, then the test value. You now have them flipped, passing in the test value followed by the expected value.

`assertThat` are "hamcrest" assertions.  They have a perhaps two or three key advantages over standard Junit assertions:
- The API takes in a "String" and a Matcher. There is no way to flip the two parameters.
- They read in a way where it is clear what the expected value is, and what the test value is, ie:
`assertThat( testValue, is(expectedValue))`
- The failure message from hamcrest assertions are better compared to standard Junit assertions - better diff output, better formatting, more intelligent use of toString
- assertion failure messages read more naturally in hamcrest (or at least I find they do):
`assertThat("we expect this to match for reasons xyz", testValue, is(matchValue))`, compared to:
`assertEquals("we expect this to match for reasons xyz", matchValue, testValue)`
- hamcrest allows for custom matchers, which allows for a much more powerful matching logic that can cut down on redundant test code (greater sharing), and remove some tedious error checking from the test code
- hamcrest has a big library of matchers, stuff to do partial string matching, collection matching, and more..

I regret that this conversion appears to have gone the wrong way. Would be better IMO to have converted to hamcrest style assertions instead of converting to standard assertions.